metrics/metricstest/resource_metrics_test.go

```diff
@@ -315,3 +319,53 @@ func TestMetricShortcuts(t *testing.T) {
 		}
 	}
 }
+
+func TestMetricFetch(t *testing.T) {
+	count := stats.Int64("count", "Test count metric", stats.UnitBytes)
+	tagKey := tag.MustNewKey("tag")
+	countView := &view.View{
+		Measure:     count,
+		Aggregation: view.Sum(),
+		TagKeys:     []tag.Key{tagKey},
+	}
+
+	view.Register(countView)
+
+	ctx, err := tag.New(context.Background(), tag.Upsert(tagKey, "alpha"))
+	if err != nil {
+		t.Errorf("Unable to create context: %v", err)
+	}
+	stats.Record(ctx, count.M(5))
+	stats.Record(ctx, count.M(3))
+
+	AssertMetric(t, Metric{Name: "count"})
```

Comment:
Can we use `AssertMetricExists(t, "count")` here?

Reply:
Yep; I picked AssertMetrics because the code seemed like it would be slightly more complex, but now I realize I could make AssertMetricExists call through to AssertMetric (done).

Reply:
This does make the error slightly more opaque ("Wrong metric <comparison>", rather than "couldn't find metric %q"), but that seems acceptable for a test helper.